plugins/net_plugin/net_plugin.cpp

```diff
@@ -1590,7 +1625,7 @@ namespace eosio {
 
          // if closing the connection we are currently syncing from, then reset our last requested and next expected.
          if( c == sync_source ) {
-            reset_last_requested_num(g);
```

Comment:
I think the purpose for this passing a reference to a unique_lock to `reset_last_requested_num()` was to ensure that a lock was taken before `sync_last_requested_num` was updated.
While it is still true that the lock is taken before this variable is reset, maybe in the future a code copy/past will be done where this is not true anymore.
Did you think that this "insurance" was not worth it?